### ISSUE TITLE
SonarScanner hangs during analysis

### ISSUE THREAD

**Issue by jvanoostveen** [SonarSource/SonarTS#194]
Our project contains about 600 files, and consists of 4 project modules (slices of the `src` and `test` directory). When running the SonarScanner (locally), it completes 2 of the modules, but hangs working on the third. Disabling this one, only makes it hangs on the next one.

```
INFO: Rules execution
INFO: Running rule analysis for `C:\Users\user\Projects\Frontend\test\tsconfig.json` with 233 files
```

I've kept it running for quite a long time, but it doesn't seem to be doing anything (no cpu usage).

I also tried running the SonarScanner with the -X option, but it doesn't show any helpful information on what it is doing. I tried applying less rules to no avail. Only limiting the total amount of files seems to have some affect, thus the total files of _all_ modules, making smaller modules doesn't seem to have any effect.
I have set the Quality Profile to the 'Sonar way recommended' variant/setting in SonarQube, but the default also doesn't work.

Is there a way to enable an even deeper debug mode to discover what is going wrong?

Environment:
Windows 10
PowerShell
Latest SonarScanner
SonarQube 6.4 (in Docker container, running locally)

**Comment by vilchik-elena** [comment]
How long did you wait? The full compilation of each file is done and for ~200 files it might take time.
Try to wait ~10min.
If not successful try these steps and tell how it goes: 
 
1. Try to run this command (from dir containing working directory which is .sonar by default):
```
node .sonar\sonarts-core\node_modules\tslint\bin\tslint --config .sonar\sonarts-core\tslint.json --format json --type-check --project C:\Users\user\Projects\Frontend\test\tsconfig.json --force
```
2. Try analysis without any rules (create a new empty profile) - in case that some rule has infinite loop
3. Execute the command from `1` point again (this time it will use new tslint.json file which has no rules activated)
4. Limit your analysis to one module with hanging rule execution. Clean the working directory (.sonar dir by default). Run analysis and during it hangs check file `sonartsRules.out` in this dir. Is it there? what is inside it?

**Comment by vilchik-elena** [comment]
Probably relates to #200

**Comment by jvanoostveen** [comment]
I did wait a long time, and I was trying to reproduce it again, but I had to split the modules up anyway because of #193. Now it did pass the scanner (in <4 min for 600+ files). We did fix a lot of issues found by tslint in our earlier runs, so maybe the code was now so clean, it is no longer an issue for our codebase.

**Comment by jvanoostveen** [comment]
Hmm after testing locally, it seem to work, but now we have it running on the build server and it is taking forever again, probably hangs...

**Comment by jamieooms** [comment]
Is there any work-around for this issue? I cannot use the plugin because it always hangs. My sonartsRules.out is empty..

**Comment by inverno** [comment]
Probably Fixed by #200 
@jvanoostveen and @jamieooms if you can, try the RC we just published : https://github.com/SonarSource/SonarTS/releases/tag/1.1.0-RC1
